cloud/cluster/cluster.go

```diff
@@ -0,0 +1,64 @@
+package cluster
+
+import (
+	"fmt"
+	"io"
+)
+
+type NodeId string
+
+type Node interface {
+	Id() NodeId
+}
+
+type NodeUpdateType int
+
+const (
+	NodeAdded NodeUpdateType = iota
+	NodeRemoved
+)
+
+// NodeUpdate represents a change to the cluster
+type NodeUpdate struct {
+	UpdateType NodeUpdateType
+	Id         NodeId
+	Node       Node // Only set for adds
+}
+
+func (u *NodeUpdate) String() string {
+	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+}
+
+// Cluster represents a cluster of Nodes.
+type Cluster interface {
+	// Members returns the current members, or an error if they can't be determined.
+	Members() ([]Node, error)
```

Comment:
Does it need to return an error? Either we know of some nodes or it doesnt and returns none. 

I agree that the code reading from service discovery needs to handle errors, but I don't think that should leak all the way to the consumers and force them to handle those errors.